tools/serverpod_cli/test/generator/dart/server_code_generator/database_repository_class_test.dart

```diff
@@ -0,0 +1,453 @@
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:serverpod_cli/src/generator/dart/server_code_generator.dart';
+import 'package:serverpod_cli/src/test_util/builders/class_definition_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/compilation_unit_helpers.dart';
+import 'package:test/test.dart';
+import 'package:path/path.dart' as path;
+
+const projectName = 'example_project';
+final config = GeneratorConfigBuilder().withName(projectName).build();
+const generator = DartServerCodeGenerator();
+
+void main() {
+  var testClassName = 'Example';
+  var testClassFileName = 'example';
+  var expectedFilePath =
+      path.join('lib', 'src', 'generated', '$testClassFileName.dart');
+
+  group('Given a class with table name when generating code', () {
+    var tableName = 'example_table';
+    var entities = [
+      ClassDefinitionBuilder()
+          .withFileName(testClassFileName)
+          .withTableName(tableName)
+          .build()
+    ];
+
+    var codeMap = generator.generateSerializableEntitiesCode(
+      entities: entities,
+      config: config,
+    );
+
+    var compilationUnit = parseString(content: codeMap[expectedFilePath]!).unit;
+
+    test('then a class named ${testClassName}AttachRepository is NOT generated',
+        () {
+      expect(
+        CompilationUnitHelpers.hasClassDeclaration(
+          compilationUnit,
+          name: '${testClassName}AttachRepository',
+        ),
+        isFalse,
+        reason:
+            'The class ${testClassName}AttachRepository was found but was expected to not exist.',
+      );
+    });
+
+    test('then a class named ${testClassName}DetachRepository is NOT generated',
```

Comment:
hmm it could make sense maybe, they should indeed never be generated on the client, but it is also outside the scope of the client. Do we currently have any tests checking that we are not creating table classes in the client? If we do it might make sense to add this if not I think we can skip.